### DISCUSSION TITLE
Why use bootstrap with Gutenberg block editor?

### DISCUSSION THREAD

**Discussion by LukaszJaro** [holger1411/understrap#1527] [General]
These days I'm really leveraging the block editor and adding custom styles to them when necessary using css grid or flexbox.

There's a few ways to use bootstrap with gutenberg like adding additional classes to blocks or re-creating entire sets of custom blocks using bootstrap markup.

Alot of core blocks work well and can be extended and this is done very easily with grid or flexbox. Many layouts can now be created in the editor, full page editing will make that even more apparent. **How are you guys using bootstrap in a useful/practical way with gutenberg?** or why would someone use understrap as a starter theme?

I'm just very curious what the professionals are doing, it's been a complicated WP landscape for past years and full site editing will throw another curve ball. I haven't used any recent understrap version so apologies if there was some integration done with bootstrap and gutenberg block editor.

**Comment by bacoords** [comment]
Great question. 

We've made sure that Understrap supports Gutenberg, including 'wide' and 'cover' style blocks. We've also pulled in the Bootstrap color palette so that you can assign colors in Gutenberg that actually match your theme's designs. And we've added some of the default typography details from the theme, so headings, tables, etc get some "bootstrap" styles.

Internally, we've been exploring some next step ideas, including theme.json support, bundling block patterns, and maybe even custom blocks that leverage Bootstrap.js (accordions, for example). 

Understrap is a developer-first framework and many of the full site editing feature are very end-user focused, so we're trying to find out where the best balance is.

**Comment by LukaszJaro** [comment]
Hi Brian, thanks for getting back! -always enjoy reading your material on understrap and here.

So there is some basic integration with Gutenberg. I was thinking same thing about Bootstrap components that can be ported over to Gutenberg, although there's actually a open ticket in gutenberg to include a accordion in the block editor!

Full site editing may even have alot of developer benefits, i.e., having different header and footer configurations easily swappable(right, left, centre aligned), some pages may want a different header/footer and all will be consistent with the Gutenberg experience which should make it simpler for end users.

Right now the value proposition of using understrap for myself is low, I much prefer framework agnostic theme since I don't load a bunch of unnecessary scripts/styles and I can apply CSS to core blocks which are already using grid/flexbox.

I'll continue watching for changes and see where the strap goes!
